pkg/client/client_test.go

```diff
@@ -157,6 +158,34 @@ func TestDependencyGraph(t *testing.T) {
 	)
 }
 
+func TestCyclicDependency(t *testing.T) {
+	testDir := getTestDir("test_cyclic_dependency")
+	assert.Equal(t, utils.DirExists(filepath.Join(testDir, "aaa")), true)
+	assert.Equal(t, utils.DirExists(filepath.Join(testDir, "aaa/kcl.mod")), true)
+	assert.Equal(t, utils.DirExists(filepath.Join(testDir, "bbb")), true)
+	assert.Equal(t, utils.DirExists(filepath.Join(testDir, "bbb/kcl.mod")), true)
+
+	pkg_path := filepath.Join(testDir, "aaa")
+
+	kpmcli, err := NewKpmClient()
+	assert.Equal(t, err, nil)
+	kclPkg, err := kpmcli.LoadPkgFromPath(pkg_path)
+	assert.Equal(t, err, nil)
+
+	currentDir, err := os.Getwd()
+	assert.Equal(t, err, nil)
+	err = os.Chdir(pkg_path)
+	assert.Equal(t, err, nil)
```

Comment:
these are done as without it an error is thrown at https://github.com/kcl-lang/kpm/blob/296b6ed4d3658fd9b2701820eed6b5b64a9c0f54/pkg/utils/utils.go#L31

in this function call, `dir` is set to `../b` and it is complaning it can't find the file as it is looking with respect to client folder instead of aaa package folder

Reply:
PR conflcted.

Reply:
Solved the merge conflict